src/NLog/Config/LoggingConfigurationFileLoader.cs

```diff
@@ -235,22 +252,84 @@ public IEnumerable<string> GetDefaultCandidateConfigFilePaths(string fileName)
 
             if (fileName == null)
             {
-                // Current config file with .config renamed to .nlog
-                string configurationFile = LogFactory.CurrentAppDomain?.ConfigurationFile;
-                if (!StringHelpers.IsNullOrWhiteSpace(configurationFile))
+                // Scan for process specific nlog-files
+                foreach (var filePath in GetAppSpecificNLogLocations(entryAssemblyLocation))
+                    yield return filePath;
+            }
+
+            foreach (var filePath in GetPrivateBinPathNLogLocations(baseDirectory, nlogConfigFile, platformFileSystemCaseInsensitive ? nLogConfigFileLowerCase : string.Empty))
+                yield return filePath;
+
+#if !SILVERLIGHT && !NETSTANDARD1_0
+            if (fileName == null)
+            {
+                // Get path to NLog.dll.nlog only if the assembly is not in the GAC
+                var nlogAssembly = typeof(LogFactory).GetAssembly();
+                if (!string.IsNullOrEmpty(nlogAssembly?.Location) && !nlogAssembly.GlobalAssemblyCache)
```

Comment:
Guess we should to do this one also (make mockable)

Reply:
Think I have the code-coverage I need. For another time.

Reply:
Fair enough